core/templates/domain/topic/TopicObjectFactory.ts

```diff
@@ -228,6 +236,15 @@ export class Topic {
         'Topic url fragment should not be longer than ' +
         `${topicUrlFragmentCharLimit} characters.`);
     }
+    if (this._skillsForDiagnosticTest.length === 0) {
+      issues.push(
+        'Skill for the diagnostic test in the topic should not be empty.');
+    }
+    if (this._skillsForDiagnosticTest.length > 3) {
+      issues.push(
+        'At most 3 skills should be added for the diagnostic test in ' +
```

Comment:
The diagnostic test for the topic should test at most 3 skills.

Reply:
Updated